### PR TITLE
Patch food

### PR DESCRIPTION
**Pull request by JaxJafinPapau**
- [x] Wrote Tests
- [x] Implemented
- [] Reviewed

# Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

## Type of change
- [x] New feature
- [] Bug Fix

# Implements/Fixes: PATCH Food route
## Description of Changes: Creates test, in foods route file creates patch route that validates whether the body attributes are present and that it can find the target resource via the params. Returns the updated resource with 200 if successful, and a 400 and error message if not.

closes #

# Check the correct boxes
- [x] This broke nothing
- [] This broke some stuff
- [] This broke everything

# Testing Changes
- [] No Tests have been changed
- [x] Some Tests have been changed
- [] All of the Tests have been changed (Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: routes/api/v1/foods.js [hunk 1]
```diff
@@ -44,4 +44,29 @@ router.get('/:id', async function(req, res, next){
   }
 })
 
+router.patch('/:id', async function(req, res, next){
+  try {
+    let food = await Food.findByPk(req.params.id)
+    let name = req.body.food.name
+    let calories = req.body.food.calories
+    if(name, calories == undefined ) {
```
Comment (wipegup) [inline routes/api/v1/foods.js]: I think this should be: ` if (name == undefined || calories == undefined)`

Tried this syntax out in a repl, and didn't seem to work.

Of note, the test only tries not passing in calories, which is maybe why this appears to work

File: routes/api/v1/foods.js [hunk 2]
```diff
@@ -44,4 +44,29 @@ router.get('/:id', async function(req, res, next){
...
+      if(new_food != null) {
+        res.setHeader(...defaultHeader);
+        res.status(200).send(JSON.stringify(new_food))
+      } else {
+        reject();
+      }
+    }
+  } catch {
+    res.setHeader(...defaultHeader);
+    res.status(400).send({ error: "Invalid ID or name/calories missing."})
```
Comment (wipegup) [inline routes/api/v1/foods.js]: How / where does this go to the catch when an invalid ID is passed?

**Review by wipegup — CHANGES_REQUESTED** [review]
A Couple of things I'm a unclear on; and one which appears to be non-functional

File: routes/api/v1/foods.js (hunk 1 above)
Comment (JaxJafinPapau) [reply routes/api/v1/foods.js]: This has been corrected. Indeed, my attempt at being fancy failed silently.

File: routes/api/v1/foods.js (hunk 2 above)
Comment (JaxJafinPapau) [reply routes/api/v1/foods.js]: Catch has been updated and error handling robustified.

**Comment by JaxJafinPapau** [comment]
Also added were tests to confirm that if the food is not found the correct error is thrown as well as scenarios where either param is missing.

**Review by wipegup — APPROVED** [review]
Looks good

**Pull request merged**